### ISSUE TITLE
ASP.NET Core RC2 Web Application with Authentication

### ISSUE THREAD

**Issue by rustd** [OmniSharp/generator-aspnet#694]
This is a close to final sample for RC2. Attaching the sample below
[WebApp_Auth.zip](https://github.com/OmniSharp/generator-aspnet/files/260050/WebApp_Auth.zip)

.NET CLI
 .NET Command Line Tools (1.0.0-rc2-002673) or higher
 Install Location: https://github.com/dotnet/cli#installers-and-binaries

NuGet Feed
https://www.myget.org/F/aspnetrelease/api/v3/index.json
https://api.nuget.org/v3/index.json

//cc @sayedihashimi

**Comment by peterblazejewicz** [comment]
this one fails on build:
> dotnet build
Project WebApplication (.NETCoreApp,Version=v1.0) will be compiled because expected outputs are missing
Compiling WebApplication for .NETCoreApp,Version=v1.0
/Users/piotrblazejewicz/development/WebApplication/project.json(13,50): warning NU1007: Dependency specified was Microsoft.AspNetCore.Mvc >= 1.0.0-rc2-20801 but ended up with Microsoft.AspNetCore.Mvc 1.0.0-rc2-20825.
/Users/piotrblazejewicz/development/WebApplication/project.json(34,81): warning NU1007: Dependency specified was Microsoft.Extensions.Configuration.EnvironmentVariables >= 1.0.0-rc2-20801 but ended up with Microsoft.Extensions.Configuration.EnvironmentVariables 1.0.0-rc2-20825.
/Users/piotrblazejewicz/development/WebApplication/project.json(35,65): warning NU1007: Dependency specified was Microsoft.Extensions.Configuration.Json >= 1.0.0-rc2-20801 but ended up with Microsoft.Extensions.Configuration.Json 1.0.0-rc2-20825.
/Users/piotrblazejewicz/development/WebApplication/project.json(36,72): warning NU1007: Dependency specified was Microsoft.Extensions.Configuration.UserSecrets >= 1.0.0-rc2-20801 but ended up with Microsoft.Extensions.Configuration.UserSecrets 1.0.0-rc2-20828.
/Users/piotrblazejewicz/development/WebApplication/project.json(38,62): warning NU1007: Dependency specified was Microsoft.Extensions.Logging.Console >= 1.0.0-rc2-20801 but ended up with Microsoft.Extensions.Logging.Console 1.0.0-rc2-20825.
/Users/piotrblazejewicz/development/WebApplication/project.json(39,60): warning NU1007: Dependency specified was Microsoft.Extensions.Logging.Debug >= 1.0.0-rc2-20801 but ended up with Microsoft.Extensions.Logging.Debug 1.0.0-rc2-20825.
/Users/piotrblazejewicz/development/WebApplication/project.json(9,69): error NU1001: The dependency Microsoft.AspNetCore.Authentication.Cookies >= 1.0.0-rc2-20801 could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(10,58): error NU1001: The dependency Microsoft.AspNetCore.Diagnostics >= 1.0.0-rc2-20801 could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(11,78): error NU1001: The dependency Microsoft.AspNetCore.Diagnostics.EntityFrameworkCore >= 1.0.0-rc2-20801 could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(12,75): error NU1001: The dependency Microsoft.AspNetCore.Identity.EntityFrameworkCore >= 1.0.0-rc2-20801 could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(14,42): error NU1001: The dependency Microsoft.AspNetCore.Razor.Tools >= 1.0.0-rc2-20801 could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(20,58): error NU1001: The dependency Microsoft.AspNetCore.StaticFiles >= 1.0.0-rc2-20801 could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(21,62): error NU1001: The dependency Microsoft.EntityFrameworkCore.Sqlite >= 1.0.0-rc2-20801 could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(22,45): error NU1001: The dependency Microsoft.EntityFrameworkCore.Tools >= 1.0.0-rc2-20801 could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(26,51): error NU1001: The dependency Microsoft.Extensions.CodeGeneration.Tools >= 1.0.0-rc2-20801 could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(30,49): error NU1001: The dependency Microsoft.Extensions.CodeGenerators.Mvc >= 1.0.0-rc2-20801 could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(40,72): error NU1001: The dependency Microsoft.VisualStudio.Web.BrowserLink.Loader >= 14.0.0-rc2-20801 could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(13,50): error NU1001: The dependency Microsoft.AspNetCore.DataProtection >= 1.0.0-rc2-20825 could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(13,50): error NU1001: The dependency Microsoft.AspNetCore.Http.Abstractions >= 1.0.0-rc2-20825 could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(13,50): error NU1001: The dependency Microsoft.AspNetCore.WebUtilities >= 1.0.0-rc2-20825 could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(13,50): error NU1001: The dependency Microsoft.Extensions.ObjectPool >= 1.0.0-rc2-20825 could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(13,50): error NU1001: The dependency Microsoft.Extensions.Logging.Abstractions >= 1.0.0-rc2-20825 could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(18,68): error NU1001: The dependency Microsoft.Extensions.Logging.Abstractions >= 1.0.0-rc2-20825 could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(19,61): error NU1001: The dependency Microsoft.Extensions.Logging.Abstractions >= 1.0.0-rc2-20825 could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(37,54): error NU1001: The dependency Microsoft.Extensions.Logging.Abstractions >= 1.0.0-rc2-20825 could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(38,62): error NU1001: The dependency Microsoft.Extensions.Logging.Abstractions >= 1.0.0-rc2-20825 could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(39,60): error NU1001: The dependency Microsoft.Extensions.Logging.Abstractions >= 1.0.0-rc2-20825 could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(13,50): error NU1001: The dependency Microsoft.Extensions.Logging.Abstractions >= 1.0.0-rc2-20801 could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(18,68): error NU1001: The dependency Microsoft.Extensions.Logging.Abstractions >= 1.0.0-rc2-20801 could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(19,61): error NU1001: The dependency Microsoft.Extensions.Logging.Abstractions >= 1.0.0-rc2-20801 could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(37,54): error NU1001: The dependency Microsoft.Extensions.Logging.Abstractions >= 1.0.0-rc2-20801 could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(38,62): error NU1001: The dependency Microsoft.Extensions.Logging.Abstractions >= 1.0.0-rc2-20801 could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(39,60): error NU1001: The dependency Microsoft.Extensions.Logging.Abstractions >= 1.0.0-rc2-20801 could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(13,50): error NU1001: The dependency Microsoft.AspNetCore.Http.Extensions >= 1.0.0-rc2-20825 could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(18,68): error NU1001: The dependency Microsoft.AspNetCore.Http.Extensions >= 1.0.0-rc2-20825 could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(19,61): error NU1001: The dependency Microsoft.AspNetCore.Http.Extensions >= 1.0.0-rc2-20825 could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(13,50): error NU1001: The dependency Microsoft.AspNetCore.Http.Extensions >= 1.0.0-rc2-20801 could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(18,68): error NU1001: The dependency Microsoft.AspNetCore.Http.Extensions >= 1.0.0-rc2-20801 could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(19,61): error NU1001: The dependency Microsoft.AspNetCore.Http.Extensions >= 1.0.0-rc2-20801 could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(18,68): error NU1001: The dependency Microsoft.AspNetCore.Hosting.Abstractions >= 1.0.0-rc2-20801 could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(19,61): error NU1001: The dependency Microsoft.AspNetCore.Hosting.Abstractions >= 1.0.0-rc2-20801 could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(13,50): error NU1001: The dependency Microsoft.AspNetCore.Hosting.Abstractions >= 1.0.0-rc2-20801 could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(18,68): error NU1001: The dependency Microsoft.AspNetCore.Hosting.Abstractions >= 1.0.0-rc2-20825 could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(19,61): error NU1001: The dependency Microsoft.AspNetCore.Hosting.Abstractions >= 1.0.0-rc2-20825 could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(13,50): error NU1001: The dependency Microsoft.AspNetCore.Hosting.Abstractions >= 1.0.0-rc2-20825 could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(18,68): error NU1001: The dependency Microsoft.AspNetCore.Hosting.Server.Abstractions >= 1.0.0-rc2-20801 could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(19,61): error NU1001: The dependency Microsoft.AspNetCore.Hosting.Server.Abstractions >= 1.0.0-rc2-20801 could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(18,68): error NU1001: The dependency Microsoft.AspNetCore.Http >= 1.0.0-rc2-20801 could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(19,61): error NU1001: The dependency Microsoft.AspNetCore.Http >= 1.0.0-rc2-20801 could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(13,50): error NU1001: The dependency Microsoft.AspNetCore.Http >= 1.0.0-rc2-20801 could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(18,68): error NU1001: The dependency Microsoft.AspNetCore.Http >= 1.0.0-rc2-20825 could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(19,61): error NU1001: The dependency Microsoft.AspNetCore.Http >= 1.0.0-rc2-20825 could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(13,50): error NU1001: The dependency Microsoft.AspNetCore.Http >= 1.0.0-rc2-20825 could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(18,68): error NU1001: The dependency Microsoft.Extensions.DependencyInjection >= 1.0.0-rc2-20801 could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(19,61): error NU1001: The dependency Microsoft.Extensions.DependencyInjection >= 1.0.0-rc2-20801 could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(13,50): error NU1001: The dependency Microsoft.Extensions.DependencyInjection >= 1.0.0-rc2-20801 could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(18,68): error NU1001: The dependency Microsoft.Extensions.DependencyInjection >= 1.0.0-rc2-20825 could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(19,61): error NU1001: The dependency Microsoft.Extensions.DependencyInjection >= 1.0.0-rc2-20825 could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(13,50): error NU1001: The dependency Microsoft.Extensions.DependencyInjection >= 1.0.0-rc2-20825 could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(18,68): error NU1001: The dependency Microsoft.Extensions.PlatformAbstractions >= 1.0.0-rc2-20801 could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(19,61): error NU1001: The dependency Microsoft.Extensions.PlatformAbstractions >= 1.0.0-rc2-20801 could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(13,50): error NU1001: The dependency Microsoft.Extensions.PlatformAbstractions >= 1.0.0-rc2-20801 could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(38,62): error NU1001: The dependency Microsoft.Extensions.PlatformAbstractions >= 1.0.0-rc2-20801 could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(18,68): error NU1001: The dependency Microsoft.Extensions.PlatformAbstractions >= 1.0.0-rc2-20825 could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(19,61): error NU1001: The dependency Microsoft.Extensions.PlatformAbstractions >= 1.0.0-rc2-20825 could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(13,50): error NU1001: The dependency Microsoft.Extensions.PlatformAbstractions >= 1.0.0-rc2-20825 could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(38,62): error NU1001: The dependency Microsoft.Extensions.PlatformAbstractions >= 1.0.0-rc2-20825 could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(13,50): error NU1001: The dependency Microsoft.Extensions.Caching.Memory >= 1.0.0-rc2-20825 could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(13,50): error NU1001: The dependency Microsoft.Net.Http.Headers >= 1.0.0-rc2-20825 could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(13,50): error NU1001: The dependency Microsoft.AspNetCore.Diagnostics.Abstractions >= 1.0.0-rc2-20825 could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(13,50): error NU1001: The dependency Microsoft.Extensions.WebEncoders >= 1.0.0-rc2-20825 could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(35,65): warning NU1012: Dependency conflict. Microsoft.Extensions.Configuration.UserSecrets 1.0.0-rc2-20828 expected Microsoft.Extensions.Configuration.Json >= 1.0.0-rc2-20828 but got 1.0.0-rc2-20825

Compilation failed.
    7 Warning(s)
    65 Error(s)

**Comment by rustd** [comment]
Can you please try with CLI version 1.0.0-rc2-002673?

**Comment by rustd** [comment]
I am uploading the final template for RC2. It has the new package versions.
[WebApp_Auth.zip](https://github.com/OmniSharp/generator-aspnet/files/263592/WebApp_Auth.zip)


Please use the following feed and CLI

**.NET CLI**
.NET Command Line Tools 1.0.0-preview1-002702
Install Location: https://github.com/dotnet/cli#installers-and-binaries

**NuGet Feed**
https://www.myget.org/F/aspnetrc2/api/v3/index.json
https://api.nuget.org/v3/index.json

**Comment by peterblazejewicz** [comment]
Updates are added to PR, though the restore and build still fails using updated feed and 1.0.0-preview1-002702:
```
Errors in /Users/piotrblazejewicz/development/WebApplication/project.json
    Unable to resolve 'NuGet.Versioning (>= 3.5.0-beta-final)' for '.NETCoreApp,Version=v1.0'.
    Unable to resolve 'NuGet.Packaging (>= 3.5.0-beta-final)' for '.NETCoreApp,Version=v1.0'.
    Unable to resolve 'NuGet.Frameworks (>= 3.5.0-beta-final)' for '.NETCoreApp,Version=v1.0'.
    Unable to resolve 'NuGet.Versioning (>= 3.5.0-beta-final)' for '.NETCoreApp,Version=v1.0'.
    Unable to resolve 'NuGet.Packaging (>= 3.5.0-beta-final)' for '.NETCoreApp,Version=v1.0'.
    Unable to resolve 'NuGet.Frameworks (>= 3.5.0-beta-final)' for '.NETCoreApp,Version=v1.0'.
    Unable to resolve 'NuGet.Packaging (>= 3.5.0-beta-final)' for '.NETCoreApp,Version=v1.0'.
    Unable to resolve 'NuGet.Versioning (>= 3.5.0-beta-final)' for '.NETCoreApp,Version=v1.0'.
    Unable to resolve 'NuGet.Frameworks (>= 3.5.0-beta-final)' for '.NETCoreApp,Version=v1.0'.
    Unable to resolve 'NuGet.Versioning (>= 3.5.0-beta-final)' for '.NETCoreApp,Version=v1.0'.
    Unable to resolve 'NuGet.Packaging (>= 3.5.0-beta-final)' for '.NETCoreApp,Version=v1.0'.
    Unable to resolve 'NuGet.Frameworks (>= 3.5.0-beta-final)' for '.NETCoreApp,Version=v1.0'.
    Unable to resolve 'NuGet.Versioning (>= 3.5.0-beta-final)' for '.NETCoreApp,Version=v1.0'.
    Unable to resolve 'NuGet.Packaging (>= 3.5.0-beta-final)' for '.NETCoreApp,Version=v1.0'.
    Unable to resolve 'NuGet.Frameworks (>= 3.5.0-beta-final)' for '.NETCoreApp,Version=v1.0'.

NuGet Config files used:
    /Users/piotrblazejewicz/development/WebApplication/NuGet.Config
    /Users/piotrblazejewicz/.nuget/NuGet/NuGet.Config

Feeds used:
    https://api.nuget.org/v3/index.json
    https://www.myget.org/F/aspnetrc2/api/v3/index.json

Installed:
    90 package(s) to /Users/piotrblazejewicz/development/WebApplication/project.json
```
```
dotnet build
Project WebApplication (.NETCoreApp,Version=v1.0) will be compiled because the version or bitness of the CLI changed since the last build
Compiling WebApplication for .NETCoreApp,Version=v1.0
/Users/piotrblazejewicz/development/WebApplication/project.json(9,69): error NU1001: The dependency Microsoft.AspNetCore.Authentication.Cookies >= 1.0.0-rc2-final could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(10,58): error NU1001: The dependency Microsoft.AspNetCore.Diagnostics >= 1.0.0-rc2-final could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(11,78): error NU1001: The dependency Microsoft.AspNetCore.Diagnostics.EntityFrameworkCore >= 1.0.0-rc2-final could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(12,75): error NU1001: The dependency Microsoft.AspNetCore.Identity.EntityFrameworkCore >= 1.0.0-rc2-final could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(13,50): error NU1001: The dependency Microsoft.AspNetCore.Mvc >= 1.0.0-rc2-final could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(14,42): error NU1001: The dependency Microsoft.AspNetCore.Razor.Tools >= 1.0.0-preview1-final could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(18,68): error NU1001: The dependency Microsoft.AspNetCore.Server.IISIntegration >= 1.0.0-rc2-final could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(19,61): error NU1001: The dependency Microsoft.AspNetCore.Server.Kestrel >= 1.0.0-rc2-final could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(20,58): error NU1001: The dependency Microsoft.AspNetCore.StaticFiles >= 1.0.0-rc2-final could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(21,62): error NU1001: The dependency Microsoft.EntityFrameworkCore.Sqlite >= 1.0.0-rc2-final could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(22,45): error NU1001: The dependency Microsoft.EntityFrameworkCore.Tools >= 1.0.0-preview1-final could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(26,81): error NU1001: The dependency Microsoft.Extensions.Configuration.EnvironmentVariables >= 1.0.0-rc2-final could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(30,62): error NU1001: The dependency Microsoft.Extensions.Logging.Console >= 1.0.0-rc2-final could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(31,60): error NU1001: The dependency Microsoft.Extensions.Logging.Debug >= 1.0.0-rc2-final could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(32,72): error NU1001: The dependency Microsoft.VisualStudio.Web.BrowserLink.Loader >= 14.0.0-rc2-final could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(33,57): error NU1001: The dependency Microsoft.VisualStudio.Web.CodeGeneration.Tools >= 1.0.0-preview1-final could not be resolved.
/Users/piotrblazejewicz/development/WebApplication/project.json(37,55): error NU1001: The dependency Microsoft.VisualStudio.Web.CodeGenerators.Mvc >= 1.0.0-preview1-final could not be resolved.

Compilation failed.
    0 Warning(s)
    17 Error(s)

Time elapsed 00:00:00.0443855
```
Does it look like packages not yet uploaded to NuGet, doesn't?

**Comment by rustd** [comment]
Those packages are missing on the rc2 feed. The team is looking into it. For now you can add this feed as well https://www.myget.org/F/nugetbuild/api/v3/index.json

**Comment by rustd** [comment]
The packages are uploaded to this feed https://www.myget.org/F/aspnetrc2/api/v3/index.json

**Comment by peterblazejewicz** [comment]
OK. The bits in PRs are updated. I'll look asap tomorrow!

**Comment by peterblazejewicz** [comment]
Updated/merged

Some reservation related to SQLite not-built-in migration: #713 

closed via: 1a5da21
